src/auth/googleauth.ts

```diff
@@ -303,14 +303,14 @@ export class GoogleAuth<T extends AuthClient = JSONClient> {
   ): Promise<ADCResponse> {
     // If we've already got a cached credential, just return it.
     if (this.cachedCredential) {
+      this.cachedCredential.setQuotaProjectIdFromEnvironment();
```

Comment:
Is this needed? If it's already cached, it should be returned as is (at least this is my understanding).

Reply:
I was on the fence for this. If we are not updating the cached credential, then the user will be confused why setting the environment variable didn't work (if they set the env var in an existing workload).
But I am ok to remove the change to cached cred as well.

Reply:
`return await this.prepareADCResponse(cachedCredential);` would look pretty clear here.

It would unify the logic and make things consistent in this method - I think in the case where the quota has been set or changed after caching customers would want it more likely than not.